cohttp-eio/src/cohttp_eio.mli

```diff
@@ -96,3 +100,81 @@ module Server : sig
 
   val not_found_handler : handler
 end
+
+(** [Client] is a HTTP/1.1 client. *)
+module Client : sig
+  type response = Http.Response.t * Eio.Buf_read.t
+
+  type host = string * int option
+  (** Represents a server host - as ip address or domain name - and an optional
+      port value, e.g. www.example.org:8080, www.reddit.com *)
+
+  type resource_path = string
+  (** Represents HTTP request resource path, e.g. "/shop/purchase",
+      "/shop/items", "/shop/categories/" etc. *)
+
+  type 'a body_disallowed_call =
+    ?version:Http.Version.t ->
+    ?headers:Http.Header.t ->
+    conn:(#Eio.Flow.two_way as 'a) ->
+    host ->
+    resource_path ->
+    response
+  (** [body_disallowed_call] denotes HTTP client calls where a request is not
+      allowed to have a request body. *)
+
+  type 'a body_allowed_call =
+    ?version:Http.Version.t ->
+    ?headers:Http.Header.t ->
+    ?body:Body.t ->
+    conn:(#Eio.Flow.two_way as 'a) ->
+    host ->
+    resource_path ->
+    response
```

Comment:
This is minor, but it might be worth putting the `body` argument first. Then it could be defined more simply as:
```suggestion
  type 'a body_allowed_call = ?body:Body.t -> 'a body_disallowed_call
```
(possibly renaming the types for clarity)

Reply:
Yes, I did consider such a scheme but found that when reading such types it read as `... so a body_allowed_call is also a body_disallowed_call`.  It didn't quite roll off the tongue for me so I left it as it was even though it is a bit repetitive. Also I didn't want to introduce another level of indirection since the types are mostly for alias and clarity for the values below, i.e get, post, head, etc.